runtime/src/snapshot_utils.rs

```diff
@@ -3549,4 +3628,73 @@ mod tests {
                 .starts_with("invalid full snapshot slot file size"));
         }
     }
+
+    #[test_case(0)]
+    #[test_case(1)]
+    #[test_case(10)]
+    fn test_serialize_deserialize_account_storage_entries(num_storages: u64) {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let bank_snapshot_dir = temp_dir.path();
+        let snapshot_slot = num_storages + 1 as Slot;
+
+        // Create AccountStorageEntries
+        let mut snapshot_storages = Vec::new();
+        for i in 0..num_storages {
+            let storage = Arc::new(AccountStorageEntry::new(
+                &PathBuf::new(),
+                i,        // Incrementing slot
+                i as u32, // Incrementing id
+                1024,
+                AccountsFileProvider::AppendVec,
+                StorageAccess::File,
+            ));
+            snapshot_storages.push(storage);
+        }
+
+        // write obsolete accounts to snapshot
+        write_obsolete_accounts_to_snapshot(bank_snapshot_dir, &snapshot_storages, snapshot_slot)
+            .unwrap();
+
+        // Deserialize
+        let deserialized_accounts =
+            deserialize_obsolete_accounts(bank_snapshot_dir, MAX_OBSOLETE_ACCOUNTS_FILE_SIZE)
+                .unwrap();
+
+        // Verify
+        for storage in &snapshot_storages {
+            assert!(deserialized_accounts.remove(&storage.slot()).unwrap().2 == 0);
+        }
+    }
+
+    #[test]
+    fn test_serialize_obsolete_accounts_too_large_file() {
```

Comment:
Should we have a similar test for deserializing an obsolete accounts file that is too large?